--- src/swft/swft_import.cpp	2006-07-20 15:57:17.000000000 +0200
+++ src/swft/swft_import.cpp.new	2009-07-04 04:09:19.000000000 +0200
@@ -5,30 +5,32 @@
 #include <string.h>
 
 void swft_addFileName( xmlNodePtr node, const char *filename ) {
-	char *name, *b;
-	const char *basename;
+	char *name, *b, *basename;
 	int l;
-	
+
+        char *real_filename = strdup(filename);
+
 	// figure basename (filename without path)
-	b = strrchr( filename, '/' );
-	basename = b ? b+1 : filename;
-	
+	b = strrchr( real_filename, '/' );
+	basename = b ? b+1 : real_filename;
+
 	l = strlen(basename);
 	name = new char[l];
 	strncpy( name, basename, l );
-	
+
 	// separate extension (set position of last . as end of string)
 	b = strrchr( name, '.' );
 	if( b ) b[0]=0;
-		
+
 	xmlSetProp( node, (const xmlChar *)"name", (const xmlChar *)name );
-	
+
 	delete[] name;
+        delete[] real_filename;
 }
 
 void swft_addData( xmlNodePtr node, char *data, int sz ) {
 	char *tmpstr = new char[ (sz * 3) ];
-	
+
 	int l = base64_encode( tmpstr, data, sz );
 	if( l > 0 ) {
 		tmpstr[l] = 0;
